src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/InstanceFailoverGroupTests.ps1

```diff
@@ -0,0 +1,255 @@
+	Assert-NotNull $fg.ManagedInstancePairs "`$fg.ManagedInstancePairs ($message)"
+	Assert-AreEqual $name $fg.Name "`$fg.Name ($message)"
+	Assert-AreEqual $rg $fg.ResourceGroupName "`$fg.ResourceGroupName ($message)"
+	Assert-AreEqual $rg $fg.PartnerResourceGroupName "`$fg.PartnerResourceGroupName ($message)"
+	Assert-AreEqual $role $fg.ReplicationRole "`$fg.ReplicationRole ($message)"
+	Assert-AreEqual $partnerRole $fg.PartnerRegion.ReplicationRole "`$fg.PartnerRegion.ReplicationRole ($message)"
+	Assert-AreEqual $failoverPolicy $fg.ReadWriteFailoverPolicy "`$fg.ReadWriteFailoverPolicy ($message)"
+	Assert-AreEqual $gracePeriod $fg.FailoverWithDataLossGracePeriodHours "`$fg.FailoverWithGracePeriodHours ($message)"
+	Assert-AreEqual $readOnlyFailoverPolicy $fg.ReadOnlyFailoverPolicy "`$fg.ReadOnlyFailoverPolicy ($message)"
+	Assert-AreEqual $true @('CATCH_UP', 'SUSPENDED', 'SEEDING').Contains($fg.ReplicationState) "`$fg.ReplicationState ($message)"
+}
+
+function Assert-InstanceFailoverGroupsEqual($expected, $actual, $role = $null, $failoverPolicy = $null, $gracePeriod = $null, $readOnlyFailoverPolicy = $null, $message = "no context provided")
+{
+	$failoverPolicy = if ($failoverPolicy -eq $null) { $expected.ReadWriteFailoverPolicy } else { $failoverPolicy }
+	$gracePeriod = if ($gracePeriod -eq $null -and $failoverPolicy -ne "Manual") { $expected.FailoverWithDataLossGracePeriodHours } else { $gracePeriod }
+	$readOnlyFailoverPolicy = if ($readOnlyFailoverPolicy -eq $null) { $expected.ReadOnlyFailoverPolicy } else { $readOnlyFailoverPolicy }
+	$role = if ($role -eq $null) { $expected.ReplicationRole } else { $role }
+
+	$partnerRole = if ($role -eq "Primary") { "Secondary" } else { "Primary" }
+
+	Validate-InstanceFailoverGroup `
+		$expected.ResourceGroupName `
+		$expected.Name `
+		$role `
+		$partnerRole `
+		$failoverPolicy `
+		$gracePeriod `
+		$readOnlyFailoverPolicy `
+		$actual `
+		$message
+}
+
+function Validate-InstanceFailoverGroupWithGet($fg, $message = "no context provided")
+{
+	$actual = $fg | Get-AzureRmSqlDatabaseInstanceFailoverGroup
+	Assert-InstanceFailoverGroupsEqual $fg $actual -message $message
+}
+
+<#
+	.SYNOPSIS
+	Tests create and update a failover group
+#>
+
+function Test-CreateInstanceFailoverGroup-Named()
+{
+	Handle-InstanceFailoverGroupTest {
+		Param($location, $partnerRegion, $rg, $managedInstanceName, $partnerManagedInstanceName)
+
+        $fgName = Get-FailoverGroupName
+		$fg = New-AzureRmSqlDatabaseInstanceFailoverGroup -Name $fgName -Location $location -ResourceGroupName $rg -PrimaryManagedInstanceName $managedInstanceName -PartnerRegion $partnerRegion -PartnerResourceGroupName $rg -PartnerManagedInstanceName $partnerManagedInstanceName
+		Validate-InstanceFailoverGroup $rg $fgName Primary Secondary Automatic 1 Disabled $fg
+		Validate-InstanceFailoverGroupWithGet $fg
+
+		Remove-AzureRmSqlDatabaseInstanceFailoverGroup -Name $fgName -Location $location -ResourceGroupName $rg -Force
```

Comment:
I'm using existing resources. There is no easy way to create a Managed Instance because the ring needs to be resized every time and that is a manual process at the moment. The resize can take about half a day.